### PR TITLE
Independent samples update

### PR DESCRIPTION
**Pull request by runjin326**
#### What scientific question is your analysis addressing?
Change "secondary" to "relapse" to only capture `tumor_descriptor == c("Recurrence", "Recurrent Tumor","Recurrent tumor","Progressive","Progressive Disease Post-Mortem")`
Additional rnaseq independent sample lists are generated

#### What was your approach?

Added `relapse_descs <- c("Recurrence", "Recurrent Tumor","Recurrent tumor","Progressive","Progressive Disease Post-Mortem")`
and use this as a filter

#### What GitHub issue does your pull request address?

PediatricOpenTargets/ticket-tracker#75
### Directions for reviewers. Tell potential reviewers what kind of feedback you are soliciting.
Please double check all the modifications
#### Which areas should receive a particularly close look?

Please check the logics in calling rnaseq independent samples - currently:
`independent-specimens.rnaseq.primary.tsv` is matched primary + RNA only primary
`independent-specimens.rnaseq.relapse.tsv` is matched relapse + RNA only relapse

#### Is the analysis in a mature enough form that the resulting figure(s) and/or table(s) are ready for review?

Yes

### Results
See ./results
#### What types of results are included (e.g., table, figure)?
.tsv files

#### What is your summary of the results?

`wc -l *.tsv 11580 independent-specimens.rnaseq.primary-plus.tsv 11043 independent-specimens.rnaseq.primary.tsv 252 independent-specimens.rnaseq.relapse.tsv 1127 independent-specimens.wgs.primary-plus.tsv 999 independent-specimens.wgs.primary.tsv 168 independent-specimens.wgs.relapse.tsv 168 independent-specimens.wgs.secondary.tsv 1865 independent-specimens.wgswxspanel.primary-plus.tsv 1735 independent-specimens.wgswxspanel.primary.tsv 173 independent-specimens.wgswxspanel.relapse.tsv 173 independent-specimens.wgswxspanel.secondary.tsv 29283 total`

#### Reproducibility Checklist

- [ ] The dependencies required to run the code in this pull request have been added to the project Dockerfile.
- [ ] This analysis has been added to continuous integration.

#### Documentation Checklist

- [ ] This analysis module has a `README` and it is up to date.
- [ ] This analysis is recorded in the table in `analyses/README.md` and the entry is up to date.
- [ ] The analytical code is documented and contains comments.

### CODE REVIEW

File: analyses/independent-samples/02-generate-independent-rnaseq.R [hunk 1]
```diff
@@ -72,6 +75,29 @@ independent_dna_sample_df <- read_tsv(opts$independent_dna_sample_df)
 
 # Filter to only samples from tumors, where composition is known to be Solid Tissue
 # for all RNA samples
+
+independent_rna_primary <- sample_df %>%
+  filter(sample_type == "Tumor", 
+         composition == "Solid Tissue"
+  ) %>%
+  independent_rna_samples(independent_dna_sample_df =
```
Comment (kgaonkar6) [inline analyses/independent-samples/02-generate-independent-rnaseq.R:83]: I recently noticed that are there are 1149  biospecimen where sample_type=="Tumor" and composition =="Bone Marrow"  should we update to use "Solid Tissue" and "Bone Marrow" here @jharenza ?

File: analyses/independent-samples/independent_rna_samples.R [hunk 2]
```diff
@@ -24,19 +24,20 @@
 #' Opetions are "primary" to select only primary/initial tumors. Primary tumors are defined as those designated "Initial CNS Tumor"+ "Primary Tumor".
 #' "primary_plus" if you would like to select other non-initial tumor RNA-Seq sample if no 
 #' initial tumor RNA-Seq sample exists
-#' or "Diagnosis" in the `tumor_descriptor` field.
+#' or "relapse" in select only relapse tumors.
```
Comment (kgaonkar6) [inline analyses/independent-samples/independent_rna_samples.R]:
```suggestion
#' or "relapse" to select relapse tumors. Relapse tumors are defined as those designated by "Recurrence", "Recurrent Tumor","Recurrent tumor","Progressive","Progressive Disease Post-Mortem" in `tumor_descriptor` field
```

**Review by kgaonkar6 — COMMENTED** [review]
Looks good to me just some minor edits and 1 question before I can approve. 

Also not related to this PR, (we didn't find this issue in OpenPBTA where I only run this module on RUN_FOR_SUBTYPING =1 ) can you update the bash script here: 
https://github.com/PediatricOpenTargets/OpenPedCan-analysis/blob/0b5099f94038e12a7e1d98cc65b0801bca8ded63/analyses/independent-samples/run-independent-samples.sh#L19
to 
`if [[ RUN_FOR_SUBTYPING -eq 0 ]]`

I had incorrectly coded `if [[ RUN_FOR_SUBTYPING == "0" ]]` instead of `-eq` for integer comparison.

File: analyses/independent-samples/02-generate-independent-rnaseq.R (hunk 1 above)
Comment (jharenza) [reply analyses/independent-samples/02-generate-independent-rnaseq.R:83]: yes please

**Review by kgaonkar6 — APPROVED** [review]
Thanks @runjin326  looks good to me!

**Review by jharenza — APPROVED** [review]
thanks for this update - looks good!

**Pull request merged**